nodejs/week3/homework/warmup/index.js

```diff
@@ -3,4 +3,145 @@ const app = express();
 
 app.get("/", (req, res) => res.send("nodejs week3 homework"));
 
-app.listen(3000, () => console.log(`Calculator:listening on port 3000`));
+//addition
+//with get
+app.get("/numbers/add", (req, res) => {
+  const first = parseInt(req.query.first);
+  const second = req.query.second;
+  let sum = first;
+  if (Array.isArray(req.query.second)) {
```

Comment:
Close!
The thing is that there could be many query parameters, instead of the second being an array. So, consider this to get all of the query parameters in one statement:
```
const values = Object.values(req.query);
```
Then, you either do a for loop as you did, or you could make use of the reduce function:
```
values.reduce((a,b)=>(parseInt(a) + parseInt(b)))
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce